diff --git a/vpr/src/base/vpr_api.cpp b/vpr/src/base/vpr_api.cpp
index e1613fdc2..d93583642 100644
--- a/vpr/src/base/vpr_api.cpp
+++ b/vpr/src/base/vpr_api.cpp
@@ -388,6 +388,8 @@ bool vpr_flow(t_vpr_setup& vpr_setup, t_arch& arch) {
     tbb::global_control c(tbb::global_control::max_allowed_parallelism, vpr_setup.num_workers);
 #endif
 
+    g_vpr_ctx.mutable_logic_levels().levelize();
+
     { //Pack
         bool pack_success = vpr_pack_flow(vpr_setup, arch);
 
