diff --git a/vpr/src/base/vpr_context.h b/vpr/src/base/vpr_context.h
index 73dbc8b04..8dcdb6844 100644
--- a/vpr/src/base/vpr_context.h
+++ b/vpr/src/base/vpr_context.h
@@ -33,6 +33,8 @@
 #include "noc_traffic_flows.h"
 #include "noc_routing.h"
 
+#include "rsbe_utils.h" // rsbe::Levelized logic_levels_
+
 /**
  * @brief A Context is collection of state relating to a particular part of VPR
  *
@@ -638,6 +640,9 @@ class VprContext : public Context {
     const PackingMultithreadingContext& packing_multithreading() const { return packing_multithreading_; }
     PackingMultithreadingContext& mutable_packing_multithreading() { return packing_multithreading_; }
 
+    const rsbe::Levelized& logic_levels() const { return logic_levels_; }
+    rsbe::Levelized& mutable_logic_levels() { return logic_levels_; }
+
   private:
     DeviceContext device_;
 
@@ -655,6 +660,13 @@ class VprContext : public Context {
     NocContext noc_;
 
     PackingMultithreadingContext packing_multithreading_;
+
+    /**
+     * @brief The Levelized class represents a graph that has been levelized, allowing efficient traversal and analysis.
+     * It provides methods to perform graph levelization, check if the graph has been levelized, and access information about
+     * nodes in each logic level and the total number of logic levels.
+     */
+    rsbe::Levelized logic_levels_;
 };
 
 #endif
